### PR TITLE
Feature/manage event

### PR DESCRIPTION
**Pull request by Keith-Go**
## Ticker
- [Manage event](https://trello.com/c/pcqJHKeC/2-qu%E1%BA%A3n-l%C3%BD-event)

## To do
- Show list event
- Show detail event page
- Edit event

## Video

https://github.com/luke-nguyen-goldenowl/join_me/assets/152830147/7a31a8a9-08f0-4800-bd44-ef162ab71dbe

### CODE REVIEW

File: lib/src/features/add_event/view/add_event_view.dart [hunk 1]
```diff
@@ -29,61 +29,79 @@ class AddEventPage extends StatelessWidget {
...
-            if (state.isPosting)
-              Container(
+            ],
+          ),
+        ),
+        BlocBuilder<AddEventBloc, AddEventState>(
+          buildWhen: (previous, current) =>
+              previous.currentPage != current.currentPage,
+          builder: ((context, state) {
+            if (state.isPosting) {
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/add_event/view/add_event_view.dart:90]: add `isPosting` to buildWhen

File: lib/src/features/manage_event/edit_event/view/edit_event_view.dart [hunk 2]
```diff
@@ -63,12 +70,39 @@ class EditEventPage extends StatelessWidget {
...
             ],
           ),
-        );
-      }),
+        ),
+        BlocBuilder<EditEventBloc, EditEventState>(
+          buildWhen: (previous, current) =>
+              previous.currentPage != current.currentPage,
+          builder: ((context, state) {
+            if (state.isSaving) {
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/manage_event/edit_event/view/edit_event_view.dart:91]: add `isSaving` into buildWhen

File: lib/src/features/manage_event/logic/manage_event_bloc.dart [hunk 3]
```diff
@@ -1,20 +1,44 @@
...
   ManageEventBloc() : super(ManageEventState(data: MPagination<MEvent>()));
 
   DomainManager get domain => DomainManager();
-
+  final MUser user = GetIt.I<AccountBloc>().state.user;
   @override
-  Future<MResult<MPaginationResponse>> get getDataAPI async {
-    await Future.delayed(const Duration(seconds: 2));
+  Future<MResult<MPaginationResponse<MEvent>>> get getDataAPI async {
+    final data =
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/manage_event/logic/manage_event_bloc.dart]: Update pagination and repository/ reference

File: lib/src/network/data/event/event_reference.dart [hunk 4]
```diff
@@ -33,6 +33,18 @@ class EventReference extends BaseCollectionReference<MEvent> {
     }
   }
 
+  Future<MResult<MEvent>> updateEvent(MEvent event) async {
```
Comment (luke-nguyen-goldenowl) [inline lib/src/network/data/event/event_reference.dart:37]: Use only `set` method

**Comment by Keith-Go** [comment]
resolved

**Comment by luke-nguyen-goldenowl** [comment]
Approved

**Pull request merged**